### ISSUE TITLE
Asset reconciliation sensor triggers runs for partitioned assets without a partition key 

### ISSUE THREAD

**Issue by slopp** [dagster-io/dagster#13566]
### Dagster version

1.2.6

### What's the issue?

The asset reconciliation sensor is creating runs for partitioned assets without supplying a partition key, causing those assets to fail. 

Sensor Ticks:
https://hooli.dagster.cloud/data-eng-prod/locations/data-eng-pipeline/sensors/freshness_sla_sensor 

Example of Incorrect Run:
https://hooli.dagster.cloud/data-eng-prod/runs/3f41a534-e886-4e18-a256-e363632b01dc?logFileKey=krmthvve

Interestingly the sensor does also launch the correct runs with the appropriate partition keys supplied.

### What did you expect to happen?

The sensor should only launch runs with appropriate partition keys.

### How to reproduce?

Run: https://github.com/dagster-io/hooli-data-eng-pipelines

Turn on the freshness_sla_sensor

For clarity I recommend updating the hourly partition definitions to be close to the repro date to avoid an overwhelming flood of backfill runs, or manually run a backfill prior to enabling the sensor

### Deployment type

Dagster Cloud

### Deployment details

Hybrid w/ k8s 

### Message from the maintainers

Impacted by this issue? Give it a 👍! We factor engagement into prioritization.

**Comment by sryza** [comment]
That link to the hooli repo isn't working for me - should it be https://github.com/dagster-io/hooli-data-eng-pipelines?

**Comment by sryza** [comment]
I ran this locally and observed this behavior. Most runs had the correct tags.  I.e. I observed a bunch of runs that targeted partitioned assets and a bunch of runs that targeted non-partitioned assets, and the partitioned ones had partition tags.

However I noticed one run that included only partitioned assets (5 of them, including raw_data/orders) and had no partition tag.

@OwenKephart - are you able to look into this when you get back?
